connection.go

```diff
@@ -510,6 +512,10 @@ func (c *Connection) Reply(ctx context.Context, opErr error) {
 	noResponse := c.kernelResponse(outMsg, inMsg.Header().Unique, op, opErr)
 
 	if !noResponse {
+		// writev is not atomic
```

Comment:
What do you mean? Why can there only be one writev syscall at a time? Does it only affect writev() or also writeMessage()

Reply:
I don't know about Linux implementation of writev but on macOS writev is definitely not atomic, meaning if there are two concurrent writev() calls the data might be written in the wrong order. I added a mutex because I routinely hit this issue during my testing. write() on the other hand is fully atomic.